arch/arm64/src/common/arm64_arch.h

```diff
@@ -0,0 +1,569 @@
+ */
+
+#define CNTV_CTL_ENABLE_BIT         BIT(0)
+#define CNTV_CTL_IMASK_BIT          BIT(1)
+
+/*  Maximum numbers of translation tables
+ *      This option specifies the maximum numbers of translation tables
+ *  excluding the base translation table. Based on this, translation
+ *  tables are allocated at compile time and used at runtime as needed.
+ *  If the runtime need exceeds preallocated numbers of translation
+ *  tables, it will result in assert. Number of translation tables
+ *  required is decided based on how many discrete memory regions
+ *  (both normal and device memory) are present on given platform and
+ *  how much granularity is required while assigning attributes
+ *  to these memory regions.
+ */
+
+#define CONFIG_MAX_XLAT_TABLES      7
+
+/* Virtual address space size
+ * Allows choosing one of multiple possible virtual address
+ * space sizes. The level of translation table is determined by
+ * a combination of page size and virtual address space size.
+ *
+ * The choice could be: 32, 36, 42, 48
+ */
+
+#define CONFIG_ARM64_VA_BITS        36
+
+/* Physical address space size
+ * Choose the maximum physical address range that the kernel will support.
+ *
+ * The choice could be: 32, 36, 42, 48
+ */
+
+#define CONFIG_ARM64_PA_BITS        36
+
+#define L1_CACHE_SHIFT		(6)
+#define L1_CACHE_BYTES		BIT(L1_CACHE_SHIFT)
+
+/****************************************************************************
+ * Type Declarations
+ ****************************************************************************/
+
+#ifdef CONFIG_ARCH_FPU
+
+/****************************************************************************
+ * armv8 fpu registers and context
+ ****************************************************************************/
+
+struct fpu_reg
+{
+  __int128 q[32];
+  uint32_t fpsr;
+  uint32_t fpcr;
+  uint64_t fpu_trap;
+};
+
+#endif
+
+/****************************************************************************
+ * Registers and exception context
+ ****************************************************************************/
+
+struct regs_context
+{
+  uint64_t  regs[31];  /* x0~x30 */
+  uint64_t  sp_elx;
+  uint64_t  elr;
+  uint64_t  spsr;
+  uint64_t  sp_el0;
+  uint64_t  exe_depth;
+  uint64_t  tpidr_el0;
+  uint64_t  tpidr_el1;
+};
+
+/****************************************************************************
+ * Public Function Prototypes
+ ****************************************************************************/
+
+/* See Arm® Architecture Reference Manual
+ * ARM DDI 0487E.a C6.2.81
+ */
+
+static inline void __DSB(void)
+{
+  __asm__ volatile ("dsb sy" : : : "memory");
+}
+
+/* See Arm® Architecture Reference Manual
+ * ARM DDI 0487E.a C6.2.79
+ */
+
+static inline void __DMB(void)
+{
+  __asm__ volatile ("dmb sy" : : : "memory");
+}
+
+/* See Arm® Architecture Reference Manual
+ * ARM DDI 0487E.a C6.2.96
+ */
+
+static inline void __ISB(void)
+{
+  __asm__ volatile ("isb" : : : "memory");
+}
```

Comment:
let's create `barriers.h` same as for ARM32 and rename to `ARM_DSB`, `ARM_DMB` and `ARM_ISB` and use it as defines

Reply:
I perfer to add prefix ARM64_ to define.
please check